### ISSUE TITLE
Multiresponse editor wrongly suggesting decimal weights

### ISSUE THREAD

**Issue by pkgh** [dthies/moodle-atto_cloze#23]
Hi,
A friend has just discovered that while editing a multi-response type of question the suggested grading options are too many. 

First of all, If the teacher chooses an option with decimal points, the question fails to decode at all, e.g.

 {1:MULTIRESPONSE:~%16.66667%a~%16.66667%b~%16.66667%c~%16.66667%d~%16.66667%e~%16.66667%f}

will produce the warning " One of the answers should have a score of 100% so it is possible to get full marks for this question. "

It IS possible to use weightings in the drop-down ONLY IF they do NOT use decimal points, e.g. this one compiles correctly:

NEW QUESTION {1:MULTIRESPONSE:~%50%ANSWER 3#GREAT~WRONG#BAD~%25%ANSWER 2#GREAT~%25%ANSWER 1#GREAT}. 

Interestingly, it seems to produce exactly the same effect as:

NEW QUESTION {1:MULTIRESPONSE:~%100%ANSWER 3#GREAT~WRONG#BAD~%50%ANSWER 2#GREAT~%50%ANSWER 1#GREAT}. 

The original proposal from OU was to only use correct or incorrect options here (cf. https://docs.moodle.org/36/en/OU_multiple_response_question_type) . The use of "weighting" in the selected grades is perhaps a (recent ?) "bonus". 

My suggestion is to shortlist the list of available options in the dropdown for multi-response to "Correct" and "Incorrect", and remove all the other options. If we want to keep the weighting options, we must at least exclude the grades with decimal points that fail to compile for the Embedded answer (Cloze) (in Moodle 3.6.5+ at least, where we have tested).

**Comment by dthies** [comment]
Yes, If I remember correctly, the answer is normallized by dividing by the total which why your last two examples are equivalent. It may be that dividing the fractional values causes a rounding error somehow.

**Comment by dthies** [comment]
@pkgh When I tested this I found it is a rounding error of some kind. If I user six 16.66667% values that would total to 1.00002. The reciprocal is 0.99998 plus some small amount that gets truncated. Multiplying and add evidently gives something less than one by more than the tolerance.
It is possible that adding one more digits to the fractional values would fix the problem.

**Comment by pkgh** [comment]
@dthies When I adjusted manually like this:

NEW QUESTION {1:MULTIRESPONSE:~%33.33333%ANSWER 3#GREAT~WRONG#BAD~%33.33333%ANSWER 2#GREAT~%33.33334%ANSWER 1#GREAT}.

to make sure the total would be exactly 100%, I still got:
" One of the answers should have a score of 100% so it is possible to get full marks for this question."

Co not sure this is a rounding thing..
Moodle 3.6.5+

**Comment by juacas** [comment]
This issue is tighly related to #26 .
There is a pull request for that: #29
